File: Assemblies/Protean.Tools/Enum.vb
Public Class EnumUtility


    ' .NET 4 Upgrade would provide this function natively.
    ' Try not to use too much as it's a bit ineffective.
    Shared Function TryParse(ByVal enumType As Type, ByVal value As String, ByVal ignoreCase As Boolean, ByRef output As Object) As Boolean

        Try
            output = [Enum].Parse(enumType, value, ignoreCase)
            Return True
        Catch ex As Exception
            Return False
        End Try

    End Function

End Class

File: Assemblies/Protean.Tools.vb/Enum.cs
using System;

namespace Protean.Tools
{
    public class EnumUtility
    {


        // .NET 4 Upgrade would provide this function natively.
        // Try not to use too much as it's a bit ineffective.
        public static bool TryParse(Type enumType, string value, bool ignoreCase, ref object output)
        {
            try
            {
                output = Enum.Parse(enumType, value, ignoreCase);
                return true;
            }
            catch (Exception ex)
            {
                return false;
            }
        }
    }
}